node/router/src/heartbeat.rs

```diff
@@ -123,6 +123,8 @@ pub trait Heartbeat<N: Network>: Outbound<N> {
             .get_connected_peers()
             .iter()
             .filter(|peer| !trusted.contains(&peer.ip()) && !bootstrap.contains(&peer.ip()))
+            .filter(|peer| !self.router().cache.contains_inbound_block_request(&peer.ip())) // Skip if the peer is syncing.
+            .filter(|peer| self.is_block_synced() || self.router().cache.num_outbound_block_requests(&peer.ip()) == 0) // Skip if you are syncing from this peer.
```

Comment:
Using a single call to filter() doing all the exclusion checks is likely more efficient. It may not be material given the small number of peers, but still a nice improvement.